hostplumber/pkg/utils/sriov/sriov.go

```diff
@@ -0,0 +1,309 @@
+package sriov
+
+import (
+	"fmt"
+	"hostplumber/pkg/consts"
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strconv"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+)
+
+func GetTotalVfs(devicePath string) int {
+	totalVfsFile := filepath.Join(devicePath, "sriov_totalvfs")
+	fd, err := os.Open(totalVfsFile)
+	defer fd.Close()
```

Comment:
This defer needs to be moved below the error handling, because it could otherwise lead to a NPE when an error is returned (and fd is nil)